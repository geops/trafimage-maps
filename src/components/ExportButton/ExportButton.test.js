import tarifverbundkarteLegend from '../../img/tarifverbund_legend.svg';

describe('ExportButton', () => {
  test('legend svg must be parseable/serializable by native DOMParser and XML Serializer', (done) => {
    try {
      const svgDoc = new DOMParser().parseFromString(
        tarifverbundkarteLegend,
        'application/xml',
      );
      svgDoc.documentElement.removeAttribute('width');
      svgDoc.documentElement.removeAttribute('height');
      new XMLSerializer().serializeToString(svgDoc);
      done();
    } catch (e) {
      // eslint-disable-next-line no-console
      console.error(
        'tarifverbund_legend.svg is not parseable/serializable by native parser/serializer',
      );
    }
  });
});
