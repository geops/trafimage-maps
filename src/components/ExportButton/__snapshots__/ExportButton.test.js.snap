// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExportDialog should match snapshot. 1`] = `
<div
  autodownload="false"
  className="no-css"
  disabled={true}
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  scale={1}
  style={
    Object {
      "opacity": 0.3,
      "pointerEvents": "none",
      "width": "auto",
    }
  }
  tabIndex={0}
  title="Als PNG speichern"
/>
`;
